library/src/main/java/com/datatorrent/lib/io/jms/JMSTransactionableStore.java

```diff
@@ -166,16 +175,15 @@ public void connect() throws IOException
     logger.debug("Entering connect. is in transaction: {}", inTransaction);
 
     try {
-      String queueName = getQueueName(getAppId(), getOperatorId());
-
       logger.debug("Base is null: {}", getBase() == null);
 
       if (getBase() != null) {
         logger.debug("Session is null: {}", getBase().getSession() == null);
       }
-
-      Queue queue = getBase().getSession().createQueue(queueName);
-      QueueBrowser browser = getBase().getSession().createBrowser(queue);
+      metaQueueName = getBase().getSubject() + ".metadata";
```

Comment:
Why not allow the user to also specify the name of this queue and default if not specified.